### ISSUE TITLE
Fullfillment fuction not working after  update to latest version !

### ISSUE THREAD

**Issue by BasitBulbulia** [nozzlegear/ShopifySharp#948]
This is my code ! 
```

'' <summary>
'' Set an existing Shopify Order to be complete
'' </summary>
'' <param name="orderId">The id of the Order to complete</param>
'' <returns>True if successful</returns>
Public Async Function SetOrderAsComplete(orderId As Long) As Threading.Tasks.Task(Of Boolean)
    Try
        Dim locationService = New LocationService(urlFromUser, shopAccessToken)
        'set the RetryExecutionPolicy to handle the API limits
        locationService.SetExecutionPolicy(New LeakyBucketExecutionPolicy())
        Dim locations = Await locationService.ListAsync()

        Dim fulfillmentService = New FulfillmentService(urlFromUser, shopAccessToken)
        'set the RetryExecutionPolicy to handle the API limits
        fulfillmentService.SetExecutionPolicy(New LeakyBucketExecutionPolicy())

        'update the orderId to be complete
        Dim result = Await fulfillmentService.CreateAsync(orderId, New ShopifySharp.Fulfillment With {.LocationId = locations(0).Id})

        Return True
    Catch ex As Exception
        If DoAllStockUpdate = False Then MsgBox("SetOrdersAsComplete--->" & ex.Message)
        Return False
    End Try
End Function
```
The error is :-
ListResult(Of Location)' cannot be indexed because it has no default property.

The error location is @ locations(0).Id

I am totally lost as to how I could fix this code

![error-ordercomplete](https://github.com/nozzlegear/ShopifySharp/assets/17301284/bc41defc-9b4f-4d22-8547-9fdc5da7829f)

**Comment by nozzlegear** [comment]
Hey @BasitBulbulia! I think the problem is the `LocationService.ListAsync` does not return a list directly, instead it returns this [`ListResult<Location>`](https://github.com/nozzlegear/ShopifySharp/blob/ad9ba6763f55957aeab5d3844344c47125532413/ShopifySharp/Lists/ListResult.cs) object, so you can't do `locations(0).Id`. 

You should be able to fix that error by using `locations.Items(0).Id`.

**Comment by BasitBulbulia** [comment]
I tried  that this is the errors in IDE
![error-two](https://github.com/nozzlegear/ShopifySharp/assets/17301284/d9ab5569-6e8f-41a9-b8ef-bc7d53d1d121)

**Comment by NodCaleb** [comment]
@nozzlegear 
I'm getting `ShopifySharp.ShopifyHttpException: (404 Not Found) Not Found` with new approach, here is my code:
```
var openFulfillmentOrders = await _fulfillmentOrderService.ListAsync(orderId);

var lineItems = openFulfillmentOrders.Select(fu => new LineItemsByFulfillmentOrder { FulfillmentOrderId = fu.Id.Value });

var fulfillmentShipping = new FulfillmentShipping
{
    Message = "Shipped",
    NotifyCustomer = false,
    TrackingInfo = new TrackingInfo
    {
        Number = trackingNumber,
        Url = "https://fedex.com",
        Company = "FedEx"
    },
    FulfillmentRequestOrderLineItems = lineItems
};

var fulfillment = await _fulfillmentService.CreateAsync(fulfillmentShipping);
```

**Comment by nozzlegear** [comment]
Hey @NodCaleb, which version of ShopifySharp were you on before this? The latest three versions target Shopify's API version 2026-01, so it's possible they've changed something in their rest API here.

**Comment by NodCaleb** [comment]
I'm was 5.something, but that doesn't matter, cause it didn't work either.
Do you mean that I'm doing it in a proper way, bit it's Shopify that has changed something?

**Comment by nozzlegear** [comment]
@NodCaleb Yes, exactly, especially if you were on version 5.x of ShopifySharp previously. Those all targeted versions of Shopify's API that are several years out of date by now:

| ShopifySharp version  | Shopify API version |
|-----------------------|---------------------|
| **6.26.0 and above.** | 2026-01             |
| 6.25.3 - 6.25.4       | 2025-07             |
| 6.20.0 - 6.25.2       | 2024-10             |
| 6.18.0 - 6.19.0       | 2024-07             |
| 6.15.0 - 6.17.0       | 2024-04             |
| 6.13.0 - 6.14.1       | 2024-01             |
| 6.3.0 - 6.12.2        | 2023-07             |
| 6.0.1 - 6.2.0         | 2023-01             |
| 5.19.0 - 5.19.1       | 2022-07             |
| 5.16.0 - 5.18.11      | 2022-04             |
| 5.14.0 - 5.15.0       | 2021-10             |
| 5.11.0 - 5.13.1       | 2021-07             |
| 5.8.0 - 5.10.0        | 2020-10             |
| 5.6.0 - 5.7.0         | 2020-07             |
| 5.0.0 - 5.5.0         | 2019-10             |
| 4.x and below         | None, unsupported   |

---

For your issue with the 404, I'd bet that the problem stems from the IDs being used. It's a common problem with the fulfillment order workflow, [check here](https://github.com/nozzlegear/ShopifySharp/issues/914) for the solution that another user and I worked through when they had a similar issue and let me know if it fixes it?

**Comment by NodCaleb** [comment]
Hello, @nozzlegear .

Updated the code according to the example you provided, now it lools like this:
```
var orderId = <someOrderId>;

var openFulfillmentOrders = await _fulfillmentOrderService.ListAsync(orderId);

var lineItems = openFulfillmentOrders.Select(fu => new LineItemsByFulfillmentOrder {
    FulfillmentOrderId = fu.Id.Value,
    FulfillmentRequestOrderLineItems = fu.FulfillmentOrderLineItems.Select(li => new FulfillmentRequestOrderLineItem
    {
        Id = li.Id.Value,
        Quantity = li.Quantity
    }).ToList()
}).ToList();

var fulfillmentShipping = new FulfillmentShipping
{
    Message = "Shipped",
    NotifyCustomer = false,
    TrackingInfo = new TrackingInfo
    {
        Number = "<trackingNumber>",
        Url = "https://fedex.com",
        Company = "FedEx"
    },
    FulfillmentRequestOrderLineItems = lineItems
};

var fulfillment = await _fulfillmentService.CreateAsync(fulfillmentShipping);
```

Now getting this error:
`ShopifySharp.ShopifyHttpException: (422 unknown) errors[0]: Invalid fulfillment order line item quantity requested.`
Despite the only line item in request has quantity 1.
